tests/integration/tests/test_cloudbeat_integration.py

```diff
@@ -55,6 +55,55 @@ def test_elastic_index_exists(elastic_client, match_type):
     :param match_type: Findings type for matching
     :return:
     """
+    query, sort = build_es_query(match_type)
+    start_time = time.time()
+    result = {}
+    while time.time() - start_time < CONFIG_TIMEOUT:
+        current_result = elastic_client.get_index_data(index_name=elastic_client.index,
+                                                       query=query,
+                                                       sort=sort)
+        if elastic_client.get_total_value(data=current_result) != 0:
+            allure.attach(json.dumps(elastic_client.get_doc_source(data=current_result),
+                                     indent=4,
+                                     sort_keys=True),
+                          match_type,
+                          attachment_type=allure.attachment_type.JSON)
+            result = current_result
+            break
+        time.sleep(1)
+
+    assert len(result) > 0, \
+        f"The findings of type {match_type} not found"
+
+
+@pytest.mark.pre_merge
+@pytest.mark.order(4)
+@pytest.mark.dependency(depends=["test_cloudbeat_pod_exist"])
+def test_leader_election(fixture_data, elastic_client):
```

Comment:
instead of passing fixture data you can pass k8s client and retrieve leader node information.